packages/design-system/src/components/Table/Table.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Alert from '../Alert/Alert';
+import PropTypes from 'prop-types';
+import TableCaption from './TableCaption';
+import classNames from 'classnames';
+
+function debounce(fn, ms) {
+  let timer;
+  return () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      timer = null;
+      fn.apply(this, arguments);
+    }, ms);
+  };
+}
+
+export const Table = ({
+  className,
+  responsiveTable,
+  striped,
+  scrollable,
+  scrollableAlert,
+  children,
+  ...attributeOptions
+}) => {
+  const container = useRef(null);
+  // The captionID is calculated and stored as init value of a ref.
+  // This ensures that the ID remains constant for all renders.
+  const captionID = useRef('caption-' + Math.random().toString(36).substr(2, 9));
+  const [tabIndex, setTabIndex] = useState(null);
+
+  // Use the Effect hook to listen for window resize event.
+  // React execute useEffect() on each re-render and returns a function that cleans up the event listener from the render.
+  // https://reactjs.org/docs/hooks-effect.html
+  useEffect(() => {
+    // Set `tabIndex` to `0` if the table width is wider than the viewport and Table is set for horizontal scrolling.
+    if (scrollable) {
+      const { scrollWidth, clientWidth } = container.current;
+      const scrollableActive = scrollWidth > clientWidth;
+      setTabIndex(scrollableActive ? '0' : null);
+
+      const debouncedHandleResize = debounce(function handleResize() {
+        const { scrollWidth, clientWidth } = container.current;
+        const scrollableActive = scrollWidth > clientWidth;
+        setTabIndex(scrollableActive ? '0' : null);
```

Comment:
I think this state change could also be a bit more smart, we could add an if statement to only set the state when needed

Reply:
Not sure I understand correctly, it presently checks for if `scrollable` is set, then perform the listen for window resize event.  Is this what you're suggesting we do?